src/Merchants.php

```diff
@@ -148,6 +148,9 @@ public static function update_or_create_merchant() {
 			throw new Exception( __( 'Response error when trying create a merchant or update the existing one.', 'pinterest-for-woocommerce' ), 400 );
 		}
 
+		// Clean merchants error code.
+		Pinterest_For_Woocommerce()::save_data( 'merchant_connected_diff_platform', false );
```

Comment:
I would remove it from here and add it to the handle feed registration. This way it is always cleared and only set if we get to this particular 4163 code.

First of all, we would have handled `merchant_connected_diff_platform` around the same place, thus easier to maintain. Second, it would always get cleared meaning that if the user setup would be triggering another type of error ( for whatever reason ) we would have only one error to focus on.

Reply:
@budzanowski, good point. Done.